src/scala/com/github/johnynek/bazel_deps/Normalizer.scala

```diff
@@ -55,14 +55,22 @@ object Normalizer {
           table(n).collect { case (_, Right(v)) => v }.toSet.size > 1
       }
 
-      def disambiguate(node: Node): Table =
+      def disambiguate(node: Node, visited: List[Node] = List()): Table = {
         table(node)
           .map(_._1.map(_.unversioned))
           .find(isAmbiguous) match {
             case None => fixVersion(node) // note that parents are not ambiguous
             case Some(None) => sys.error("unreachable, roots are never ambiguous")
-            case Some(Some(p)) => disambiguate(p)
+            case Some(Some(p)) => {
+              if (!visited.contains(p)) {
+                val res = disambiguate(p, p :: visited)
+                res
+              } else {
+                fixVersion(node)
+              }
+            }
           }
+      }
```

Comment:
Not actually sure this change is related to my change for `classifiers`, but I left it in case it is.

Reply:
I don't see how it is. It seems like it may be useful in general. Can we move to a separate PR (which we can probably merge sooner).

Reply:
👍 I'll pull it out into a separate PR and see if I can write a test case that covers it.

Reply:
Moved to https://github.com/johnynek/bazel-deps/pull/133